core/trino-main/src/main/java/io/trino/operator/TopNOperator.java

```diff
@@ -209,17 +229,20 @@ private void addPage(Page page)
                 return TransformationState.needsMoreData();
             }
 
-            // no more input, return results
-            Page page = null;
-            while (page == null && !topNProcessor.noMoreOutput()) {
-                page = topNProcessor.getOutput();
-            }
+            Page page = topNProcessor.getOutput();
 
             if (page != null) {
                 return TransformationState.ofResult(page, false);
             }
 
-            return TransformationState.finished();
+            if (isFlushing) {
+                isFlushing = false;
+                pageBuffer.unfinish();
```

Comment:
I'm trying to understand a bit more about the intended logic here about what can cause a pagebuffer to become "unfinished", but it seems like the only place where isFlushing can be set to true is on line 216, which doesn't seem possible to reach?